src/ruby/ext/grpc/rb_channel.c

```diff
@@ -225,14 +258,11 @@ static VALUE grpc_rb_channel_watch_connectivity_state(VALUE self,
     return Qnil;
   }
   grpc_channel_watch_connectivity_state(
-      ch,
-      (grpc_connectivity_state)NUM2LONG(last_state),
-      grpc_rb_time_timeval(deadline, /* absolute time */ 0),
-      cq,
-      tag);
+      ch, (grpc_connectivity_state)NUM2LONG(last_state),
+      grpc_rb_time_timeval(deadline, /* absolute time */ 0), cq, tag);
```

Comment:
good catch. I'm not sure if this is safe.

cc @ctiller this could potentially call `watch_connectivity_state`, on the same `grpc_channel`, on two separate completion queues, one polled on `next` and the other on `pluck`.